recipes/screen_capture_lite/all/conanfile.py

```diff
@@ -0,0 +1,132 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+required_conan_version = ">=1.52.0"
+
+class ScreenCaptureLiteConan(ConanFile):
+    name = "screen_capture_lite"
+    license = "MIT"
+    description = "cross platform screen/window capturing library "
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/smasherprog/screen_capture_lite"
+    topics = ("screen-capture", "screen-ercorder")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 20
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "11",
+            "clang": "10",
+            "apple-clang": "10",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.requires("xorg/system")
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        check_min_vs(self, 192)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support."
+                )
+
+        if self.info.settings.compiler == "clang" and self.info.settings.compiler.get_safe("libcxx") == "libstdc++":
+            raise ConanInvalidConfiguration(f"{self.ref} does not support clang with libstdc++")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_SHARED_LIBS"] = bool(self.options.shared)
```

Comment:
Usually we do https://github.com/conan-io/conan-center-index/blob/30e910738b04042a3546bc2605168339aadb95a5/recipes/json-c/all/conanfile.py#L58-L59

but this also works

Reply:
Thank you for your review.
I want to honor the standard way as possible.

Reply:
`tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"` doesn't work well on this recipe.
Because upstream CMakeLists.txt define BUILD_SHARED_LIBS as follows:
```
set(BUILD_SHARED_LIBS OFF CACHE INTERNAL "Build shared libraries")
```
So I update patch files to delete this line.

Reply:
Ahhh, might be worth upstreaming that patch 😉